File: app/src/main/java/ru/nehodov/weatherforecast/dao/DailyDao.java
package ru.nehodov.weatherforecast.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ru.nehodov.weatherforecast.entities.Daily;

@Dao
public interface DailyDao {

    @Insert
    void insertDailyForecast(Daily[] dailies);

    @Query("SELECT * FROM daily")
    LiveData<List<Daily>> getDailyForecast();

    @Query("DELETE FROM daily")
    void deleteDailyForecast();
}

File: app/src/main/java/ru/nehodov/weatherforecast/dao/DailyDao.kt
package ru.nehodov.weatherforecast.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import ru.nehodov.weatherforecast.entities.Daily

@Dao
interface DailyDao {

    @Insert
    suspend fun insert(daily: List<Daily>)

    @Query("SELECT * FROM daily")
    fun dailyForecastData(): Flow<List<Daily>>

    @Query("DELETE FROM daily")
    suspend fun deleteAll()

}